### PR TITLE
Feat/env yml

### PR DESCRIPTION
**Pull request by justinGilmer**
To fall in line with `foyer` and `mBuild`, this PR switches the
`requirements*.txt` files to environment.yml files to present
dependencies.

This also updates the documentation and Dockerfile to use these new
files.

@jennyfothergill and @umesh-timalsina can yall see if i missed anything major here?

### CODE REVIEW

File: docs/installation.rst [hunk 1]
```diff
@@ -71,11 +45,26 @@ is made to support Python 2 because it is considered obsolete as of early 2020.
...
-``GMSO`` uses ``py.test`` to execute its unit tests. To run them, first install some extra depdencies:
+``GMSO`` uses ``py.test`` to execute its unit tests. To run them, first install the dev environment
 ::
-    $ conda install --file requirements-test.txt
 
+    $ git clone https://github.com/mosdef-hub/gmso.git
+    $ cd gmso
+    $ conda env create -f environment-dev.yml
+    $ conda activate gmso-dev
+    $ pip install -e .
```
Comment (jennyfothergill) [inline docs/installation.rst:55]: I'm not sure we need to repeat these steps. Instead we might reference the section header.

File: docs/installation.rst [hunk 2]
```diff
@@ -2,60 +2,34 @@
...
 Once all dependencies are installed, the ``GMSO`` itself can be installed.
-It is currently only available through its source code. It will be available
-through ``pip`` and ``conda`` in the future.
 ::
 
     $ git clone https://github.com/mosdef-hub/gmso.git
     $ cd gmso
+    $ conda env create -f environment-dev.yml
+    $ conda activate gmso-dev
     $ pip install -e .
```
Comment (jennyfothergill) [inline docs/installation.rst:33]: I feel like these could be combined... 🤔 Maybe something like 

Installing from source
----------------------
Dependencies of ``GMSO`` are listed in the files ``environment.yml`` (lightweight environment specification containing minimal dependencies) and ``environment-dev.yml`` (comprehensive environment specification including testing packages for developers). The ``gmso`` or ``gmso-dev`` ``conda`` environments can be created with::

Comment (jennyfothergill) [reply docs/installation.rst:33]: Then once the dependencies are installed install gmso::

File: environment-dev.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+name: gmso_dev
```
Comment (jennyfothergill) [inline environment-dev.yml:1]:
```suggestion
name: gmso-dev
```

**Review by jennyfothergill — COMMENTED** [review]
I made a couple minor rewording change comments and found one typo in the dev yml.

File: docs/installation.rst (hunk 2 above)
Comment (justinGilmer) [reply docs/installation.rst]: Let me know what you think of these changes!

**Review by jennyfothergill — APPROVED** [review]
LGTM 👍

**Comment by justinGilmer** [comment]
I think these tests will continue to fail until the changes here are merged, but I cannot recall. The errors for the tests are that the `requirements.txt` files are missing

**Comment by jennyfothergill** [comment]
Yeah, I agree. I don't see any reference to requirements.txt in the workflow yml in your PR, though. It's probably running the workflow on master? Umesh might know more.

**Comment by justinGilmer** [comment]
> Yeah, I agree. I don't see any reference to requirements.txt in the workflow yml in your PR, though. It's probably running the workflow on master? Umesh might know more.

Yeah i just went to the pipeline settings and its yml it uses has not been updated yet. So it should be good once this is merged.

**Review by daico007 — APPROVED** [review]
LGTM

File: docs/installation.rst [hunk 4]
```diff
@@ -2,60 +2,43 @@
...
-    work well for most installations.
+    # for gmso-dev
+    $ conda env create -f environment-dev.yml
+    $ conda activate gmso
 
-Installing dependencies with `pip <https://pypi.org/project/pip/>`_
--------------------------------------------------------------------
-::
+    # install a non-editable version of gmso
+    $ pip install
```
Comment (daico007) [inline docs/installation.rst:31]:
```suggestion
    $ pip install .
```

**Pull request merged**